lib/winapi/sync.c

```diff
@@ -38,6 +174,240 @@ VOID LeaveCriticalSection (LPCRITICAL_SECTION lpCriticalSection)
     RtlLeaveCriticalSection(lpCriticalSection);
 }
 
+#define SRW_GLOBAL_LOCK_BIT 0x40000000
+#define SRW_READER_LOCK_BIT 0x80000000
+#define SRW_GLOBAL_LOCK_MASK (~SRW_GLOBAL_LOCK_BIT) // 0xBFFFFFFF
+#define SRW_READER_LOCK_MASK (~SRW_READER_LOCK_BIT) // 0x7FFFFFFF
+#define SRW_READER_COUNT_MASK (SRW_GLOBAL_LOCK_MASK & SRW_READER_LOCK_MASK) // 0x3FFFFFFF
```

Comment:
The `_MASK` are very weird / confusing:
- `LOCK_MASK` describe the bits that are *not* the `LOCK_BIT`.
- `COUNT_MASK` is describing the bits that *are* the `COUNT_BITS` (if it existed).

I'd suggest to only have the `_BIT` and use the well recognized `& ~FOO_BIT` and `| FOO_BIT`.
So `SRW_READER_COUNT_MASK` would also just be `SRW_READER_COUNT_BITS`. 

*(Wether you use `_MASK` or `_BIT` doesn't matter to me. My point is to have 1 of them and negate in the code where you use it)*

I feel the current way of having both is confusing because of inconsistency mentioned above; and also because you use 2 names for the same thing.

Reply:
Alright, changed.